### PR TITLE
PWGCF: Update task multiparticle-correlations-ar

### PR DESCRIPTION
**Pull request by ariedel-cern**
- Implement comments from previous pull request
- Add support for calculation Qvectors

### CODE REVIEW

File: PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx [hunk 1]
```diff
@@ -71,314 +108,366 @@ static constexpr std::string_view TrackVariableNames[kLAST_TrackVariable] = {"Pt
...
+
+    // Check if event survives event cuts, where we can get the values for the variables immediately
+    if (!AR::SurviveCut(cfgVX.value, collision.posX()) ||
+        !AR::SurviveCut(cfgVY.value, collision.posY()) ||
+        !AR::SurviveCut(cfgVZ.value, collision.posZ()) ||
+        !AR::SurviveCut(cfgVABS.value, AR::abs(collision.posX(), collision.posY(), collision.posZ())) ||
+        !AR::SurviveCut(cfgCEN.value, collision.centRun2V0M()) ||
+        !AR::SurviveCut(cfgMULNC.value, collision.numContrib()) ||
+        !AR::SurviveCut(cfgMULTPC.value, collision.multTPC())) {
+      flag = false;
```
Comment (saganatt) [inline PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx:353]:
```suggestion
      return false;
```

Comment (saganatt) [reply PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx:353]: With this, you don't need `if` in line 360

File: PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx [hunk 2]
```diff
@@ -71,314 +108,366 @@ static constexpr std::string_view TrackVariableNames[kLAST_TrackVariable] = {"Pt
...
     }
-    // cut multiplicity (tpc)
-    if (cfgEvent.at(AR::kMULTPC).value.at(5) > 0 &&
-        (collision.multTPC() < cfgEvent.at(AR::kMULTPC).value.at(3) || collision.multTPC() > cfgEvent.at(AR::kMULTPC).value.at(4))) {
-      SurviveCut = false;
+
+    // at last, check if event also passes this cut
+    if (!AR::SurviveCut(cfgMULQ.value, MultiplicityQ) ||
+        !AR::SurviveCut(cfgMULW.value, MultiplicityW)) {
+      flag = false;
```
Comment (saganatt) [inline PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx:372]:
```suggestion
    return AR::SurviveCut(cfgMULQ.value, MultiplicityQ) && AR::SurviveCut(cfgMULW.value, MultiplicityW)
```
and no need for the flag variable

File: PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx [hunk 3]
```diff
@@ -13,19 +13,56 @@
...
 using namespace o2::framework;
+using namespace o2::framework::expressions;
 
-namespace MultiParticleCorrelationsARTaskGlobalConfig
+// Define useful constants and enums in a separate name space
+namespace MultiParticleCorrelationsARTaskGlobalVariables
 {
-// setup for event variables
+// for before and after applying cuts
+enum BA {
```
Comment (saganatt) [inline PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx:39]: Maybe "BeforeAfterCuts", for more readability?

File: PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx [hunk 4]
```diff
@@ -13,19 +13,56 @@
...
+  kBEFORE,
+  kAFTER,
+  kLAST_BA
+};
+static constexpr std::string_view BeforeAfter[kLAST_BA] = {"before/", "after/"};
+
+// for reconstructed and simulated data
+enum RS {
```
Comment (saganatt) [inline PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx:47]: What about "RecoSim"?

File: PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx [hunk 5]
```diff
@@ -71,314 +108,366 @@ static constexpr std::string_view TrackVariableNames[kLAST_TrackVariable] = {"Pt
...
+    if (!(Value > ConfigValue.at(kLCUT) && Value < ConfigValue.at(kUCUT))) {
+      flag = false;
+    }
+  }
+  return flag;
+}
+}; // namespace MultiParticleCorrelationsARTaskGlobalVariables
+
+// use an alias for our namespace
+namespace AR = MultiParticleCorrelationsARTaskGlobalVariables;
```
Comment (saganatt) [inline PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx:141]:
```suggestion
using namespace AR = MultiParticleCorrelationsARTaskGlobalVariables;
```

File: PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx [hunk 6]
```diff
@@ -71,314 +108,366 @@ static constexpr std::string_view TrackVariableNames[kLAST_TrackVariable] = {"Pt
...
+        !AR::SurviveCut(cfgDCAZ.value, track.dcaZ()) ||
+        !AR::SurviveCut(cfgDCAXY.value, track.dcaXY()) ||
+        !AR::SurviveCut(cfgTPCCLUSTERS.value, track.tpcNClsFound()) ||
+        !AR::SurviveCut(cfgTPCCROSSEDROWS.value, track.tpcNClsCrossedRows()) ||
+        !AR::SurviveCut(cfgTPCCHI2.value, track.tpcChi2NCl()) ||
+        !AR::SurviveCut(cfgITSCLUSTERS.value, track.itsNCls())) {
+      flag = false;
+    }
+
+    return flag;
```
Comment (saganatt) [inline PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx:397]: You can just return AND of meeting all conditions: `return AR::SurviveCut(...) && AR::SurviveCut(...) && ...`

File: PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx (hunk 4 above)
Comment (saganatt) [reply PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx:47]: or "RecoSimEnum", when RecoSim is already taken?

**Review by saganatt — CHANGES_REQUESTED** [review]
Hi, I left some smaller suggestions for improving the code, please consider them.

File: PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx (hunk 1 above)
Comment (ariedel-cern) [reply PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx]: Fixed.

File: PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx (hunk 2 above)
Comment (ariedel-cern) [reply PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx]: Fixed.

File: PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx (hunk 3 above)
Comment (ariedel-cern) [reply PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx]: Thanks for pointing that out. More readability is always better.

File: PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx (hunk 4 above)
Comment (ariedel-cern) [reply PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx]: Fixed.

File: PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx (hunk 5 above)
Comment (ariedel-cern) [reply PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx:141]: I acutally like using the namespace explicitly. Even though its more typing, I think it is easier to read and find the stuff during debugging.

File: PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx (hunk 6 above)
Comment (ariedel-cern) [reply PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx]: Fixed.

**Comment by ariedel-cern** [comment]
Thanks saganat, for looking over my PR. I really appreciate the feedback. I hope with the fixes implemented you can give my PR a +1.

File: PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx (hunk 5 above)
Comment (saganatt) [reply PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx:141]: ah, you are right, I was thinking about `using` as creating an alias, but it does not work for namespaces in this way,

**Review by saganatt — APPROVED** [review]

**Pull request merged**